File: project-frontend/src/app/productCategory.ts
/* export interface ProductCategory {
    id: string;
    name: string;
    categoryId: string;
    productCategory: ProductCategory;
    children: ProductCategory[];
} */

export class ProductCategory {
    id: string | undefined;
    name: string  | undefined;
    categoryId: string  | undefined;
    productCategory: ProductCategory | undefined;
    children: ProductCategory[] | undefined;
}

File: project-backend/src/main/java/com/example/project/productCategory/ProductCategory.java
package com.example.project.productCategory;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import lombok.Data;

@Data
@Document(collection = "productCategory")
public class ProductCategory {
    @Id
    public String id;
    @Field
    public String name;
    @Field
    public String categoryId;
    @Field
    @DBRef
    public ProductCategory productCategory;
}
